### PR TITLE
Fix viewer mode showing incorrect grammar when graph error occurs

### LINKED ISSUE #868: Bug: Viewer mode fails to show correct grammar text when graph error occurs
### Similar issues

- [X] I have searched and found no existing similar issues

### How are you using Studio?

Legend Studio

### Current and expected behavior

During a graph building error, viewer mode stays in form mode and reports graph building error. When user tries to investigate and goes to text mode, the grammar is shown from the state of the pure model at the time of the error. Therefore viewer mode is showing an incorrect grammar. It should show the grammar using the original entities when loading. 

### Steps to reproduce

Have a project with the following grammar (consider using model loader). Create a review and merge this. Then try to view the project in viewer mode, and toggle `text-mode`. The grammar is wrong.

### Model data

```
Class model::NewClass extends model::NewClass2
{
  prop1: String[1];
}


###Mapping
Mapping model::NewMapping
(
  model::NewClass: Pure
  {
  }
)
```

### Environment

```text
Chrome
```

### Possible solution and workaround

_No response_

### Contribution

- [ ] I would like to work on the fix for this issue

### PR DESCRIPTION
**Pull request by gayathrir11**
## Summary

Closes #868 

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

***Before the fix***
pic2

***After the fix***
pic1

### CODE REVIEW

File: packages/legend-studio/src/stores/EditorStore.ts [hunk 1]
```diff
@@ -1281,6 +1281,12 @@ export class EditorStore {
         showLoading: true,
       });
       try {
+        if (this.isInViewerMode) {
+          this.graphState.buildGraph(
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/EditorStore.ts:1285]: How does this fix the problem. 
I would assume that the `buildGraph` would still break with the given entities and form an incomplete graph and then give incomplete text.

**Comment by akphi** [comment]
@gayathrir11 why don't we see the red underline in the second screenshot?

**Comment by MauricioUyaguari** [comment]
Seems like we may need to force a compilation after entering text mode to show the error

**Comment by gayathrir11** [comment]
> Seems like we may need to force a compilation after entering text mode to show the error

yeah. Looking into the code to understand how we do it for a workspace

**Comment by akphi** [comment]
Thanks @gayathrir11 and @MauricioUyaguari after discussing, I realized we are aiming at the wrong target here. What we should do is not focusing on toggling to text-mode, but try to do something similar to editor mode in viewer mode where we properly reveal compilation error in `text-mode`. I have opened https://github.com/finos/legend-studio/pull/1199 to address this, so I'll be closing this PR here.

**Pull request closed without merging**